 # **Data Preparation**



In [73]:
import pandas as pd
import numpy as np
from datetime import datetime, timedelta
import warnings
warnings.filterwarnings('ignore')

In [74]:
df = pd.read_csv("student.csv")

In [75]:
df

รหัสนักศึกษา      ชื่อ   อายุ     เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย   20.0     ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง   21.0    หญิง           วิทย์  3.45       165.0   
2          64003       NaN    NaN    Male         Science  3.67       175.0   
3          64004  สมศักดิ์   22.0       ช  คณะวิทยาศาสตร์ -1.00       168.0   
4          64005      สมใจ   19.0       ญ           วิศวะ  3.89       160.0   
5          64006     กานต์  150.0     ชาย     Engineering  3.12       172.0   
6          64007     กิตติ   21.0    หญิง        วิศวกรรม  4.50       158.0   
7          64008      จิรา   20.0       F      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป   23.0       M            Arts  3.34       163.0   
9          64010      ดารา   21.0    หญิง           ศิลป์  3.56       167.0   
10         64011      ธีระ    NaN     NaN             NaN  3.78         NaN   
11         64012       นภา   22.0     ชาย           วิทย์   NaN       171.0   
12         64013     บุญมี   20.0    หญิง           วิศวฯ  3.23       169.0   
13         64014     ปรีชา   21.0     ชาย    Liberal Arts  3.45       164.0   
14         64015     พิมพ์   22.0  Female             Sci  3.67       162.0   
15         64016      ภูมิ   19.0       ช     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี   20.0       ญ  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา   21.0    หญิง             NaN  3.23       161.0   
18         64019     รัตนา    NaN     ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา   22.0     NaN         Science  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
2          NaN                     25.0      2024-01-13  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
5         75.0                     12.0      2023-12-23  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับข้อมูลในคอลัมน์ ชื่อ ที่มีชื่อเป็น NaN โดยใช้คำสั่ง df.dropna(subset=['ชื่อ'])

* ใช้คำสั่ง df.dropna(subset=['ชื่อ']) เพื่อลบแถวทั้งหมดที่มีค่า NaN (ค่าหาย) ออกจากข้อมูลไฟล์ student.csv
* เนื่องจากคอลัมน์ "ชื่อ" เป็นข้อมูลสำคัญที่ควรมีอยู่ครบ ไม่สามารถนำค่าเฉลี่ยหรือค่าสถิติอื่นมาทดแทนได้เหมือนกับคอลัมน์อายุจึงควรลบแถวนั้นออกไป

In [76]:
df_clean = df.dropna(subset=['ชื่อ'])

In [77]:
df_original = df
df = df_clean
df

รหัสนักศึกษา      ชื่อ   อายุ     เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย   20.0     ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง   21.0    หญิง           วิทย์  3.45       165.0   
3          64004  สมศักดิ์   22.0       ช  คณะวิทยาศาสตร์ -1.00       168.0   
4          64005      สมใจ   19.0       ญ           วิศวะ  3.89       160.0   
5          64006     กานต์  150.0     ชาย     Engineering  3.12       172.0   
6          64007     กิตติ   21.0    หญิง        วิศวกรรม  4.50       158.0   
7          64008      จิรา   20.0       F      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป   23.0       M            Arts  3.34       163.0   
9          64010      ดารา   21.0    หญิง           ศิลป์  3.56       167.0   
10         64011      ธีระ    NaN     NaN             NaN  3.78         NaN   
11         64012       นภา   22.0     ชาย           วิทย์   NaN       171.0   
12         64013     บุญมี   20.0    หญิง           วิศวฯ  3.23       169.0   
13         64014     ปรีชา   21.0     ชาย    Liberal Arts  3.45       164.0   
14         64015     พิมพ์   22.0  Female             Sci  3.67       162.0   
15         64016      ภูมิ   19.0       ช     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี   20.0       ญ  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา   21.0    หญิง             NaN  3.23       161.0   
18         64019     รัตนา    NaN     ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา   22.0     NaN         Science  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
5         75.0                     12.0      2023-12-23  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับข้อมูลในคอลัมน์ อายุ ซึ่งเป็นการจัดการค่าผิดปกติ
*   ในคอลัมน์อายุมีอายุ 150 ซึ่งเป็นค่าที่ผิดปกติจึงควรตัดออกเพื่อให้ได้ข้อมูลที่ถูกต้องสมบูรณ์
#โดยทำการกรองค่าผิดปกติเพื่อตัดแถวที่มีค่าอายุเป็น 150 ปี ให้หายไป


In [78]:
Q1 = df['อายุ'].quantile(0.25)
Q3 = df['อายุ'].quantile(0.75)
IQR = Q3 - Q1

lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR

df_no_outliers = df[(df['อายุ'].isna()) | ((df['อายุ'] >= lower_bound) & (df['อายุ'] <= upper_bound))]

In [79]:
df = df_no_outliers
df

รหัสนักศึกษา      ชื่อ  อายุ     เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย  20.0     ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง  21.0    หญิง           วิทย์  3.45       165.0   
3          64004  สมศักดิ์  22.0       ช  คณะวิทยาศาสตร์ -1.00       168.0   
4          64005      สมใจ  19.0       ญ           วิศวะ  3.89       160.0   
6          64007     กิตติ  21.0    หญิง        วิศวกรรม  4.50       158.0   
7          64008      จิรา  20.0       F      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป  23.0       M            Arts  3.34       163.0   
9          64010      ดารา  21.0    หญิง           ศิลป์  3.56       167.0   
10         64011      ธีระ   NaN     NaN             NaN  3.78         NaN   
11         64012       นภา  22.0     ชาย           วิทย์   NaN       171.0   
12         64013     บุญมี  20.0    หญิง           วิศวฯ  3.23       169.0   
13         64014     ปรีชา  21.0     ชาย    Liberal Arts  3.45       164.0   
14         64015     พิมพ์  22.0  Female             Sci  3.67       162.0   
15         64016      ภูมิ  19.0       ช     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี  20.0       ญ  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา  21.0    หญิง             NaN  3.23       161.0   
18         64019     รัตนา   NaN     ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา  22.0     NaN         Science  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับค่า NaN ในคอลัมน์ อายุ โดยใช้ค่าเฉลี่ยและนำข้อมูลมาเติมค่าลงไปแทนค่าNaN

In [80]:
mean_age = df['อายุ'].mean()
df['อายุ'].fillna(mean_age, inplace=True)

In [81]:
df

รหัสนักศึกษา      ชื่อ    อายุ     เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย  20.000     ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง  21.000    หญิง           วิทย์  3.45       165.0   
3          64004  สมศักดิ์  22.000       ช  คณะวิทยาศาสตร์ -1.00       168.0   
4          64005      สมใจ  19.000       ญ           วิศวะ  3.89       160.0   
6          64007     กิตติ  21.000    หญิง        วิศวกรรม  4.50       158.0   
7          64008      จิรา  20.000       F      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป  23.000       M            Arts  3.34       163.0   
9          64010      ดารา  21.000    หญิง           ศิลป์  3.56       167.0   
10         64011      ธีระ  20.875     NaN             NaN  3.78         NaN   
11         64012       นภา  22.000     ชาย           วิทย์   NaN       171.0   
12         64013     บุญมี  20.000    หญิง           วิศวฯ  3.23       169.0   
13         64014     ปรีชา  21.000     ชาย    Liberal Arts  3.45       164.0   
14         64015     พิมพ์  22.000  Female             Sci  3.67       162.0   
15         64016      ภูมิ  19.000       ช     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี  20.000       ญ  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา  21.000    หญิง             NaN  3.23       161.0   
18         64019     รัตนา  20.875     ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา  22.000     NaN         Science  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# ทำให้ข้อมูลอายุเป็นค่าจำนวนเต็มเนื่องจากค่าอายุดูไม่ค่อยมีความเหมาะสมกับการดูข้อมูล จึงต้องการทำให้เป็นจำนวนเต็มเพื่อให้ง่ายต่อการอ่านข้อมูล

In [82]:
df['อายุ'] = np.floor(df['อายุ']).astype(int)
df

รหัสนักศึกษา      ชื่อ  อายุ     เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20     ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21    หญิง           วิทย์  3.45       165.0   
3          64004  สมศักดิ์    22       ช  คณะวิทยาศาสตร์ -1.00       168.0   
4          64005      สมใจ    19       ญ           วิศวะ  3.89       160.0   
6          64007     กิตติ    21    หญิง        วิศวกรรม  4.50       158.0   
7          64008      จิรา    20       F      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23       M            Arts  3.34       163.0   
9          64010      ดารา    21    หญิง           ศิลป์  3.56       167.0   
10         64011      ธีระ    20     NaN             NaN  3.78         NaN   
11         64012       นภา    22     ชาย           วิทย์   NaN       171.0   
12         64013     บุญมี    20    หญิง           วิศวฯ  3.23       169.0   
13         64014     ปรีชา    21     ชาย    Liberal Arts  3.45       164.0   
14         64015     พิมพ์    22  Female             Sci  3.67       162.0   
15         64016      ภูมิ    19       ช     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20       ญ  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21    หญิง             NaN  3.23       161.0   
18         64019     รัตนา    20     ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา    22     NaN         Science  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จากคอลัมน์ เพศ จะเห็นว่าข้อมูลเพศใช้ข้อมูลกระจายหลายรูปแบบมากเกินไป มีทั้ง ชาย หญิง หรือ ช ญ Female F M จึงควรทำข้อมูลให้ข้อมูลเป็นมาตรฐานเดียวกันคือใช้ ชาย และ หญิง

In [83]:
def standardize_gender(gender):
    if gender in ['ชาย', 'ช', 'Male', 'M']:
        return 'ชาย'
    elif gender in ['หญิง', 'ญ', 'Female', 'F']:
        return 'หญิง'
    else:
        return gender

df['เพศ'] = df['เพศ'].apply(standardize_gender)

In [84]:
df

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง           วิทย์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย  คณะวิทยาศาสตร์ -1.00       168.0   
4          64005      สมใจ    19  หญิง           วิศวะ  3.89       160.0   
6          64007     กิตติ    21  หญิง        วิศวกรรม  4.50       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย            Arts  3.34       163.0   
9          64010      ดารา    21  หญิง           ศิลป์  3.56       167.0   
10         64011      ธีระ    20   NaN             NaN  3.78         NaN   
11         64012       นภา    22   ชาย           วิทย์   NaN       171.0   
12         64013     บุญมี    20  หญิง           วิศวฯ  3.23       169.0   
13         64014     ปรีชา    21   ชาย    Liberal Arts  3.45       164.0   
14         64015     พิมพ์    22  หญิง             Sci  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง             NaN  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา    22   NaN         Science  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับค่า  NaN ในคอลัมน์เพศ โดยใช้การเติมค่าด้วยค่าที่พบบ่อยที่สุดลงไปแทนค่า NaN

In [85]:
df['เพศ'].fillna(df['เพศ'].mode()[0], inplace=True)

In [86]:
df

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง           วิทย์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย  คณะวิทยาศาสตร์ -1.00       168.0   
4          64005      สมใจ    19  หญิง           วิศวะ  3.89       160.0   
6          64007     กิตติ    21  หญิง        วิศวกรรม  4.50       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย            Arts  3.34       163.0   
9          64010      ดารา    21  หญิง           ศิลป์  3.56       167.0   
10         64011      ธีระ    20  หญิง             NaN  3.78         NaN   
11         64012       นภา    22   ชาย           วิทย์   NaN       171.0   
12         64013     บุญมี    20  หญิง           วิศวฯ  3.23       169.0   
13         64014     ปรีชา    21   ชาย    Liberal Arts  3.45       164.0   
14         64015     พิมพ์    22  หญิง             Sci  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง             NaN  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา    22  หญิง         Science  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับคอลัมน์ คณะ	 ทำให้ข้อมูลเป็นมาตรฐานแบบเดียวกัน

In [87]:
def standardize_faculty(name):
    if name in ['วิทยาศาสตร์', 'วิทย์', 'Science', 'คณะวิทยาศาสตร์' ,'Sci']:
        return 'วิทยาศาสตร์'
    elif name in ['ศิลปศาสตร์', 'ศิลป์', 'Arts' , 'Liberal Arts','คณะศิลปศาสตร์']:
        return 'ศิลปศาสตร์'
    elif name in ['วิศวกรรมศาสตร์', 'วิศวฯ', 'วิศวะ','วิศวกรรม','Engineering', 'คณะวิศวกรรมศาสตร์']:
        return 'วิศวกรรมศาสตร์'
    else:
        return name

# ใช้งาน
df['คณะ'] = df['คณะ'].apply(standardize_faculty)

In [88]:
df

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง     วิทยาศาสตร์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย     วิทยาศาสตร์ -1.00       168.0   
4          64005      สมใจ    19  หญิง  วิศวกรรมศาสตร์  3.89       160.0   
6          64007     กิตติ    21  หญิง  วิศวกรรมศาสตร์  4.50       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย      ศิลปศาสตร์  3.34       163.0   
9          64010      ดารา    21  หญิง      ศิลปศาสตร์  3.56       167.0   
10         64011      ธีระ    20  หญิง             NaN  3.78         NaN   
11         64012       นภา    22   ชาย     วิทยาศาสตร์   NaN       171.0   
12         64013     บุญมี    20  หญิง  วิศวกรรมศาสตร์  3.23       169.0   
13         64014     ปรีชา    21   ชาย      ศิลปศาสตร์  3.45       164.0   
14         64015     พิมพ์    22  หญิง     วิทยาศาสตร์  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง             NaN  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา    22  หญิง     วิทยาศาสตร์  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับค่า  NaN ในคอลัมน์คณะ โดยใช้การเติมค่าด้วยค่าที่พบบ่อยที่สุดลงไปแทนค่า NaN

In [89]:
df['คณะ'].fillna(df['คณะ'].mode()[0], inplace=True)

In [90]:
df

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง     วิทยาศาสตร์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย     วิทยาศาสตร์ -1.00       168.0   
4          64005      สมใจ    19  หญิง  วิศวกรรมศาสตร์  3.89       160.0   
6          64007     กิตติ    21  หญิง  วิศวกรรมศาสตร์  4.50       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย      ศิลปศาสตร์  3.34       163.0   
9          64010      ดารา    21  หญิง      ศิลปศาสตร์  3.56       167.0   
10         64011      ธีระ    20  หญิง     วิทยาศาสตร์  3.78         NaN   
11         64012       นภา    22   ชาย     วิทยาศาสตร์   NaN       171.0   
12         64013     บุญมี    20  หญิง  วิศวกรรมศาสตร์  3.23       169.0   
13         64014     ปรีชา    21   ชาย      ศิลปศาสตร์  3.45       164.0   
14         64015     พิมพ์    22  หญิง     วิทยาศาสตร์  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง     วิทยาศาสตร์  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา    22  หญิง     วิทยาศาสตร์  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับค่าผิดปกติในคอลัมน์ GPA ที่มีค่าไม่ปกติ คือ -1.00 และ 4.50	เพราะgpa ไม่สามารถ ติดลบ หรือเกิน 4.00 ได้ ทำให้เป็นค่า NaN

In [91]:
df.loc[(df['GPA'] < 0) | (df['GPA'] > 4.00), 'GPA'] = np.nan

In [92]:
df

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง     วิทยาศาสตร์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย     วิทยาศาสตร์   NaN       168.0   
4          64005      สมใจ    19  หญิง  วิศวกรรมศาสตร์  3.89       160.0   
6          64007     กิตติ    21  หญิง  วิศวกรรมศาสตร์   NaN       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย      ศิลปศาสตร์  3.34       163.0   
9          64010      ดารา    21  หญิง      ศิลปศาสตร์  3.56       167.0   
10         64011      ธีระ    20  หญิง     วิทยาศาสตร์  3.78         NaN   
11         64012       นภา    22   ชาย     วิทยาศาสตร์   NaN       171.0   
12         64013     บุญมี    20  หญิง  วิศวกรรมศาสตร์  3.23       169.0   
13         64014     ปรีชา    21   ชาย      ศิลปศาสตร์  3.45       164.0   
14         64015     พิมพ์    22  หญิง     วิทยาศาสตร์  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง     วิทยาศาสตร์  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์   NaN       174.0   
19         64020     ลัดดา    22  หญิง     วิทยาศาสตร์  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับค่า NaN ในคอลัมน์ GPA โดยใช้ค่าเฉลี่ยและนำข้อมูลมาเติมค่าลงไปแทนค่าNaN

In [93]:
mean_gpa = round(df['GPA'].mean(), 2)
df['GPA'].fillna(mean_gpa, inplace=True)

In [94]:
df

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง     วิทยาศาสตร์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย     วิทยาศาสตร์  3.44       168.0   
4          64005      สมใจ    19  หญิง  วิศวกรรมศาสตร์  3.89       160.0   
6          64007     กิตติ    21  หญิง  วิศวกรรมศาสตร์  3.44       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย      ศิลปศาสตร์  3.34       163.0   
9          64010      ดารา    21  หญิง      ศิลปศาสตร์  3.56       167.0   
10         64011      ธีระ    20  หญิง     วิทยาศาสตร์  3.78         NaN   
11         64012       นภา    22   ชาย     วิทยาศาสตร์  3.44       171.0   
12         64013     บุญมี    20  หญิง  วิศวกรรมศาสตร์  3.23       169.0   
13         64014     ปรีชา    21   ชาย      ศิลปศาสตร์  3.45       164.0   
14         64015     พิมพ์    22  หญิง     วิทยาศาสตร์  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง     วิทยาศาสตร์  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์  3.44       174.0   
19         64020     ลัดดา    22  หญิง     วิทยาศาสตร์  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับค่า NaN ในคอลัมน์ ส่วนสูง_cm โดยใช้ค่าเฉลี่ยและนำข้อมูลมาเติมค่าลงไปแทนค่าNaN

In [95]:
mean_height = df['ส่วนสูง_cm'].mean()
df['ส่วนสูง_cm'].fillna(mean_height, inplace=True)

In [96]:
df

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง     วิทยาศาสตร์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย     วิทยาศาสตร์  3.44       168.0   
4          64005      สมใจ    19  หญิง  วิศวกรรมศาสตร์  3.89       160.0   
6          64007     กิตติ    21  หญิง  วิศวกรรมศาสตร์  3.44       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย      ศิลปศาสตร์  3.34       163.0   
9          64010      ดารา    21  หญิง      ศิลปศาสตร์  3.56       167.0   
10         64011      ธีระ    20  หญิง     วิทยาศาสตร์  3.78       167.0   
11         64012       นภา    22   ชาย     วิทยาศาสตร์  3.44       171.0   
12         64013     บุญมี    20  หญิง  วิศวกรรมศาสตร์  3.23       169.0   
13         64014     ปรีชา    21   ชาย      ศิลปศาสตร์  3.45       164.0   
14         64015     พิมพ์    22  หญิง     วิทยาศาสตร์  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง     วิทยาศาสตร์  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์  3.44       174.0   
19         64020     ลัดดา    22  หญิง     วิทยาศาสตร์  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11         NaN                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับค่า NaN ในคอลัมน์ น้ำหนัก_kgb โดยใช้ค่าเฉลี่ยและนำข้อมูลมาเติมค่าลงไปแทนค่าNaN

In [97]:
mean_weight = round(df['น้ำหนัก_kg'].mean(), 1)
df['น้ำหนัก_kg'].fillna(mean_weight, inplace=True)

In [98]:
df

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง     วิทยาศาสตร์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย     วิทยาศาสตร์  3.44       168.0   
4          64005      สมใจ    19  หญิง  วิศวกรรมศาสตร์  3.89       160.0   
6          64007     กิตติ    21  หญิง  วิศวกรรมศาสตร์  3.44       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย      ศิลปศาสตร์  3.34       163.0   
9          64010      ดารา    21  หญิง      ศิลปศาสตร์  3.56       167.0   
10         64011      ธีระ    20  หญิง     วิทยาศาสตร์  3.78       167.0   
11         64012       นภา    22   ชาย     วิทยาศาสตร์  3.44       171.0   
12         64013     บุญมี    20  หญิง  วิศวกรรมศาสตร์  3.23       169.0   
13         64014     ปรีชา    21   ชาย      ศิลปศาสตร์  3.45       164.0   
14         64015     พิมพ์    22  หญิง     วิทยาศาสตร์  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง     วิทยาศาสตร์  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์  3.44       174.0   
19         64020     ลัดดา    22  หญิง     วิทยาศาสตร์  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                      NaN      2024-01-07  
11        62.9                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                      NaN      2023-12-18  
19        64.0                     23.0      2024-01-06

# จัดการกับค่า NaN ในคอลัมน์ ชั่วโมงเรียน_ต่อสัปดาห์ โดยใช้ค่าเฉลี่ยและนำข้อมูลมาเติมค่าลงไปแทนค่าNaN

In [99]:
mean_time = round(df['ชั่วโมงเรียน_ต่อสัปดาห์'].mean(), 1)
df['ชั่วโมงเรียน_ต่อสัปดาห์'].fillna(mean_time, inplace=True)

In [100]:
df

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง     วิทยาศาสตร์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย     วิทยาศาสตร์  3.44       168.0   
4          64005      สมใจ    19  หญิง  วิศวกรรมศาสตร์  3.89       160.0   
6          64007     กิตติ    21  หญิง  วิศวกรรมศาสตร์  3.44       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย      ศิลปศาสตร์  3.34       163.0   
9          64010      ดารา    21  หญิง      ศิลปศาสตร์  3.56       167.0   
10         64011      ธีระ    20  หญิง     วิทยาศาสตร์  3.78       167.0   
11         64012       นภา    22   ชาย     วิทยาศาสตร์  3.44       171.0   
12         64013     บุญมี    20  หญิง  วิศวกรรมศาสตร์  3.23       169.0   
13         64014     ปรีชา    21   ชาย      ศิลปศาสตร์  3.45       164.0   
14         64015     พิมพ์    22  หญิง     วิทยาศาสตร์  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง     วิทยาศาสตร์  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์  3.44       174.0   
19         64020     ลัดดา    22  หญิง     วิทยาศาสตร์  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  
0         65.0                     15.0      2024-01-23  
1         58.0                     20.0      2024-02-05  
3         70.0                     10.0      2023-12-30  
4         52.0                     30.0      2024-01-27  
6         48.0                     35.0      2024-01-05  
7         82.0                     18.0      2024-01-23  
8         55.0                     22.0      2024-02-11  
9         60.0                     28.0      2024-01-03  
10        68.0                     21.8      2024-01-07  
11        62.9                     16.0      2023-12-26  
12        72.0                     24.0      2023-12-26  
13        54.0                     19.0      2024-01-08  
14        56.0                     26.0      2024-02-06  
15        78.0                     14.0      2024-01-20  
16        62.0                     21.0      2024-01-24  
17        50.0                     27.0      2024-01-08  
18        76.0                     21.8      2023-12-18  
19        64.0                     23.0      2024-01-06

# **Feature Engineering**

In [101]:
data = df

#แปลงข้อมูลประเภทเป็นตัวเลขจาก True False ให้เป็น 1 และ 0

In [102]:
data_encoded = pd.get_dummies(data, columns=['คณะ'])*1

In [103]:
data_encoded

รหัสนักศึกษา      ชื่อ  อายุ   เพศ   GPA  ส่วนสูง_cm  น้ำหนัก_kg  \
0          64001     สมชาย    20   ชาย  3.21       170.0        65.0   
1          64002    สมหญิง    21  หญิง  3.45       165.0        58.0   
3          64004  สมศักดิ์    22   ชาย  3.44       168.0        70.0   
4          64005      สมใจ    19  หญิง  3.89       160.0        52.0   
6          64007     กิตติ    21  หญิง  3.44       158.0        48.0   
7          64008      จิรา    20  หญิง  2.98       180.0        82.0   
8          64009    ชนาธิป    23   ชาย  3.34       163.0        55.0   
9          64010      ดารา    21  หญิง  3.56       167.0        60.0   
10         64011      ธีระ    20  หญิง  3.78       167.0        68.0   
11         64012       นภา    22   ชาย  3.44       171.0        62.9   
12         64013     บุญมี    20  หญิง  3.23       169.0        72.0   
13         64014     ปรีชา    21   ชาย  3.45       164.0        54.0   
14         64015     พิมพ์    22  หญิง  3.67       162.0        56.0   
15         64016      ภูมิ    19   ชาย  3.89       173.0        78.0   
16         64017      มานี    20  หญิง  3.01       166.0        62.0   
17         64018      ยุพา    21  หญิง  3.23       161.0        50.0   
18         64019     รัตนา    20   ชาย  3.44       174.0        76.0   
19         64020     ลัดดา    22  หญิง  3.45       168.0        64.0   

    ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  คณะ_วิทยาศาสตร์  \
0                      15.0      2024-01-23                1   
1                      20.0      2024-02-05                1   
3                      10.0      2023-12-30                1   
4                      30.0      2024-01-27                0   
6                      35.0      2024-01-05                0   
7                      18.0      2024-01-23                0   
8                      22.0      2024-02-11                0   
9                      28.0      2024-01-03                0   
10                     21.8      2024-01-07                1   
11                     16.0      2023-12-26                1   
12                     24.0      2023-12-26                0   
13                     19.0      2024-01-08                0   
14                     26.0      2024-02-06                1   
15                     14.0      2024-01-20                1   
16                     21.0      2024-01-24                0   
17                     27.0      2024-01-08                1   
18                     21.8      2023-12-18                0   
19                     23.0      2024-01-06                1   

    คณะ_วิศวกรรมศาสตร์  คณะ_ศิลปศาสตร์  
0                    0               0  
1                    0               0  
3                    0               0  
4                    1               0  
6                    1               0  
7                    0               1  
8                    0               1  
9                    0               1  
10                   0               0  
11                   0               0  
12                   1               0  
13                   0               1  
14                   0               0  
15                   0               0  
16                   1               0  
17                   0               0  
18                   0               1  
19                   0               0

# การสร้าง Feature จาก GPA

In [104]:
data['ระดับผลการเรียนเฉลี่ยสะสม'] = pd.cut(data['GPA'],
    bins=[-float('inf'), 2.00, 3.20, float('inf')],
    labels=['รอพินิจ', 'ปกติ', 'เรียนเด่น'],
    right=False
)

In [105]:
data

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง     วิทยาศาสตร์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย     วิทยาศาสตร์  3.44       168.0   
4          64005      สมใจ    19  หญิง  วิศวกรรมศาสตร์  3.89       160.0   
6          64007     กิตติ    21  หญิง  วิศวกรรมศาสตร์  3.44       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย      ศิลปศาสตร์  3.34       163.0   
9          64010      ดารา    21  หญิง      ศิลปศาสตร์  3.56       167.0   
10         64011      ธีระ    20  หญิง     วิทยาศาสตร์  3.78       167.0   
11         64012       นภา    22   ชาย     วิทยาศาสตร์  3.44       171.0   
12         64013     บุญมี    20  หญิง  วิศวกรรมศาสตร์  3.23       169.0   
13         64014     ปรีชา    21   ชาย      ศิลปศาสตร์  3.45       164.0   
14         64015     พิมพ์    22  หญิง     วิทยาศาสตร์  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง     วิทยาศาสตร์  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์  3.44       174.0   
19         64020     ลัดดา    22  หญิง     วิทยาศาสตร์  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  \
0         65.0                     15.0      2024-01-23   
1         58.0                     20.0      2024-02-05   
3         70.0                     10.0      2023-12-30   
4         52.0                     30.0      2024-01-27   
6         48.0                     35.0      2024-01-05   
7         82.0                     18.0      2024-01-23   
8         55.0                     22.0      2024-02-11   
9         60.0                     28.0      2024-01-03   
10        68.0                     21.8      2024-01-07   
11        62.9                     16.0      2023-12-26   
12        72.0                     24.0      2023-12-26   
13        54.0                     19.0      2024-01-08   
14        56.0                     26.0      2024-02-06   
15        78.0                     14.0      2024-01-20   
16        62.0                     21.0      2024-01-24   
17        50.0                     27.0      2024-01-08   
18        76.0                     21.8      2023-12-18   
19        64.0                     23.0      2024-01-06   

   ระดับผลการเรียนเฉลี่ยสะสม  
0                  เรียนเด่น  
1                  เรียนเด่น  
3                  เรียนเด่น  
4                  เรียนเด่น  
6                  เรียนเด่น  
7                       ปกติ  
8                  เรียนเด่น  
9                  เรียนเด่น  
10                 เรียนเด่น  
11                 เรียนเด่น  
12                 เรียนเด่น  
13                 เรียนเด่น  
14                 เรียนเด่น  
15                 เรียนเด่น  
16                      ปกติ  
17                 เรียนเด่น  
18                 เรียนเด่น  
19                 เรียนเด่น

# การรวม Features ของข้อมูลน้ำหนักและส่วนสูงมาหาค่า BMI และแสดงผลสุขภาพได้นั่นเอง

In [106]:
data['BMI'] = data['น้ำหนัก_kg'] / (data['ส่วนสูง_cm']/100)**2
data['สุขภาพ'] = pd.cut(data['BMI'],
                                   bins=[0, 18.5, 25, 30, 100],
                                   labels=['ผอม', 'ปกติ', 'อ้วน', 'อ้วนมาก'])

In [107]:
data[['ชื่อ', 'BMI', 'สุขภาพ']]

ชื่อ        BMI สุขภาพ
0      สมชาย  22.491349   ปกติ
1     สมหญิง  21.303949   ปกติ
3   สมศักดิ์  24.801587   ปกติ
4       สมใจ  20.312500   ปกติ
6      กิตติ  19.227688   ปกติ
7       จิรา  25.308642   อ้วน
8     ชนาธิป  20.700817   ปกติ
9       ดารา  21.513859   ปกติ
10      ธีระ  24.382373   ปกติ
11       นภา  21.510892   ปกติ
12     บุญมี  25.209201   อ้วน
13     ปรีชา  20.077335   ปกติ
14     พิมพ์  21.338211   ปกติ
15      ภูมิ  26.061679   อ้วน
16      มานี  22.499637   ปกติ
17      ยุพา  19.289379   ปกติ
18     รัตนา  25.102391   อ้วน
19     ลัดดา  22.675737   ปกติ

In [108]:
data

รหัสนักศึกษา      ชื่อ  อายุ   เพศ             คณะ   GPA  ส่วนสูง_cm  \
0          64001     สมชาย    20   ชาย     วิทยาศาสตร์  3.21       170.0   
1          64002    สมหญิง    21  หญิง     วิทยาศาสตร์  3.45       165.0   
3          64004  สมศักดิ์    22   ชาย     วิทยาศาสตร์  3.44       168.0   
4          64005      สมใจ    19  หญิง  วิศวกรรมศาสตร์  3.89       160.0   
6          64007     กิตติ    21  หญิง  วิศวกรรมศาสตร์  3.44       158.0   
7          64008      จิรา    20  หญิง      ศิลปศาสตร์  2.98       180.0   
8          64009    ชนาธิป    23   ชาย      ศิลปศาสตร์  3.34       163.0   
9          64010      ดารา    21  หญิง      ศิลปศาสตร์  3.56       167.0   
10         64011      ธีระ    20  หญิง     วิทยาศาสตร์  3.78       167.0   
11         64012       นภา    22   ชาย     วิทยาศาสตร์  3.44       171.0   
12         64013     บุญมี    20  หญิง  วิศวกรรมศาสตร์  3.23       169.0   
13         64014     ปรีชา    21   ชาย      ศิลปศาสตร์  3.45       164.0   
14         64015     พิมพ์    22  หญิง     วิทยาศาสตร์  3.67       162.0   
15         64016      ภูมิ    19   ชาย     วิทยาศาสตร์  3.89       173.0   
16         64017      มานี    20  หญิง  วิศวกรรมศาสตร์  3.01       166.0   
17         64018      ยุพา    21  หญิง     วิทยาศาสตร์  3.23       161.0   
18         64019     รัตนา    20   ชาย      ศิลปศาสตร์  3.44       174.0   
19         64020     ลัดดา    22  หญิง     วิทยาศาสตร์  3.45       168.0   

    น้ำหนัก_kg  ชั่วโมงเรียน_ต่อสัปดาห์ วันที่ลงทะเบียน  \
0         65.0                     15.0      2024-01-23   
1         58.0                     20.0      2024-02-05   
3         70.0                     10.0      2023-12-30   
4         52.0                     30.0      2024-01-27   
6         48.0                     35.0      2024-01-05   
7         82.0                     18.0      2024-01-23   
8         55.0                     22.0      2024-02-11   
9         60.0                     28.0      2024-01-03   
10        68.0                     21.8      2024-01-07   
11        62.9                     16.0      2023-12-26   
12        72.0                     24.0      2023-12-26   
13        54.0                     19.0      2024-01-08   
14        56.0                     26.0      2024-02-06   
15        78.0                     14.0      2024-01-20   
16        62.0                     21.0      2024-01-24   
17        50.0                     27.0      2024-01-08   
18        76.0                     21.8      2023-12-18   
19        64.0                     23.0      2024-01-06   

   ระดับผลการเรียนเฉลี่ยสะสม        BMI สุขภาพ  
0                  เรียนเด่น  22.491349   ปกติ  
1                  เรียนเด่น  21.303949   ปกติ  
3                  เรียนเด่น  24.801587   ปกติ  
4                  เรียนเด่น  20.312500   ปกติ  
6                  เรียนเด่น  19.227688   ปกติ  
7                       ปกติ  25.308642   อ้วน  
8                  เรียนเด่น  20.700817   ปกติ  
9                  เรียนเด่น  21.513859   ปกติ  
10                 เรียนเด่น  24.382373   ปกติ  
11                 เรียนเด่น  21.510892   ปกติ  
12                 เรียนเด่น  25.209201   อ้วน  
13                 เรียนเด่น  20.077335   ปกติ  
14                 เรียนเด่น  21.338211   ปกติ  
15                 เรียนเด่น  26.061679   อ้วน  
16                      ปกติ  22.499637   ปกติ  
17                 เรียนเด่น  19.289379   ปกติ  
18                 เรียนเด่น  25.102391   อ้วน  
19                 เรียนเด่น  22.675737   ปกติ